src/new_skills/NewSkill.cpp

```diff
@@ -0,0 +1,136 @@
+//
+// Created by jessevw on 21.02.20.
+//
+
+#include "include/roboteam_ai/new_skills/NewSkill.h"
+#include "skills/Skill.h"
+
+#include <include/roboteam_ai/utilities/Settings.h>
+
+#include <cmath>
+
+#include "control/ControlUtils.h"
+#include "utilities/Constants.h"
+#include "utilities/GameStateManager.hpp"
+#include "utilities/RobotDealer.h"
+#include "world/Ball.h"
+#include "world/Robot.h"
+#include "world/World.h"
+
+namespace rtt::ai {
+
+    NewSkill::NewSkill(std::string name, bt::Blackboard::Ptr blackboard) : bt::Leaf(std::move(name), std::move(blackboard)) {
+        robot = std::make_shared<Robot>();
+        ball = std::make_shared<Ball>();
+    }
+
+    void NewSkill::publishRobotCommand() {
+        if (!SETTINGS.isLeft()) {
+            command = rotateRobotCommand(command);
+        }
+
+        limitRobotCommand();
+
+        if (command.id() == -1) {
+            std::cerr << "the command is trying to send a RobotCommand to id " << command.id() << std::endl;
+            if (robot && robot->id != -1) {
+                command.set_id(robot->id);
+                io::io.publishRobotCommand(command);  // We default to our robots being on the left if parameter is not set
+            }
+        } else {
+            io::io.publishRobotCommand(command);  // We default to our robots being on the left if parameter is not set
+        }
+        // refresh the robotcommand after it has been sent
+        refreshRobotCommand();
+    }
+
+    std::string NewSkill::node_name() { return name; }
+
+    NewSkill::Status NewSkill::update() {
+        robotId = -15;// TODO: after the tactic is implemented, needs to be some way the skill figures out which id to send to. Should be known by the tactic
```

Comment:
The robot will be assigned anyway by the robot dealer, right? Why not take directly the id of the robot?

Reply:
Well what should be here is that the role tells the Skill "hey you're controlling this robot, his ID is x". But I havent gotten aroudn to implementing that yet. Is that an answer to your question?